tests/integration/integration_suite_test.go

```diff
@@ -1,13 +1,187 @@
-	. "github.com/onsi/gomega"
+const (
+	medusaTestTable    = "medusa_test"
+	medusaTestKeyspace = "medusa"
 )
 
-func TestIntegration(t *testing.T) {
-	RegisterFailHandler(Fail)
-	RunSpecs(t, "Integration Suite")
+func TestMain(m *testing.M) {
+	updateHelmDeps()
+	os.Exit(m.Run())
+}
+
+func initializeCluster(t *testing.T) {
+	CheckK8sClusterIsReachable(t)
+	InstallTraefik(t)
+	CreateNamespace(t)
+	CheckK8ssandraNamespaceWasCreated(t)
+}
+
+func cleanupCluster(t *testing.T) {
+	UninstallK8ssandraHelmRelease(t)
+	UninstallTraefikHelmRelease(t)
+	DeleteK8ssandraNamespace(t)
+	DeleteTraefikNamespace(t)
+	CheckNamespaceIsAbsent(t)
+}
+
+// Reaper scenario:
+// - Install Traefik
+// - Create a namespace
+// - Register a cluster with 3 Cassandra nodes
+// - Verify that Reaper correctly initializes
+// - Start a repair on the reaper_db keyspace
+// - Wait for at least one segment to be processed
+// - Cancel the repair
+// - Terminate the namespace and delete the cluster
+func TestReaperDeploymentScenario(t *testing.T) {
+	t.Run("Initialize cluster", initializeCluster)
+
+	t.Run("Deploy cluster", deployClusterForReaper)
+	t.Run("Check expected resources are present", checkResourcePresenceForReaper)
+	t.Run("Wait for Reaper pod to be ready", waitForReaperPod)
+	t.Run("Check Reaper has registered the required elements", checkReaperRegistered)
+	t.Run("Start repair on the reaper_db keyspace", triggerRepair)
+	t.Run("Wait for at least one segment to be processed and cancel the repair", waitForSegmentDoneAndCancel)
+
+	t.Run("Tear down cluster", cleanupCluster)
+}
+
+func deployClusterForReaper(t *testing.T) {
+	DeployClusterWithValues(t, "default", "three_nodes_cluster_with_reaper.yaml")
+}
+
+func checkResourcePresenceForReaper(t *testing.T) {
+	CheckResourceIsPresent(t, "service", "app.kubernetes.io/managed-by=reaper-operator")
+	CheckClusterExpectedResources(t)
+}
+
+func waitForReaperPod(t *testing.T) {
+	WaitForReaperPod(t)
+}
+
+func checkReaperRegistered(t *testing.T) {
+	CheckKeyspaceExists(t, "reaper_db")
+	CheckClusterIsRegisteredInReaper(t, "k8ssandra")
+}
+
+func triggerRepair(t *testing.T) {
+	TriggerRepair(t, "reaper_db")
+}
+
+func waitForSegmentDoneAndCancel(t *testing.T) {
+	WaitForOneSegmentToBeDone(t)
+	CancelRepair(t)
+}
+
+// Medusa scenario (invoked with a specific backend name):
+// - Register a cluster with 1 node
+// - Potentially install backend specific dependencies (such as Minio)
+// - Create the backend credentials secret
+// - Create a keyspace and a table
+// - Load 10 rows and check that we can read 10 rows
+// - Perform a backup with Medusa
+// - Load 10 rows and check that we can read 20 rows
+// - Restore the backup
+// - Verify that we can read 10 rows
+// - Terminate the namespace and delete the cluster
+func TestMedusaDeploymentScenario(t *testing.T) {
+	backends := []string{"Minio", "S3"}
+	for _, backend := range backends {
+		t.Run(fmt.Sprintf("%s: Initialize cluster", backend), initializeCluster)
```

Comment:
Same as above, if `initializeCluster` fails, there is no point in proceeding.

Reply:
See [my previous answer](https://github.com/k8ssandra/k8ssandra/pull/544#discussion_r595794009).